tcpMaxConn.go

```diff
@@ -28,16 +32,70 @@ func runTcpConnectionsInParallel(numberConnections int, delay int, host string,
 	wg.Wait()
 }
 
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+func askForUserConfirmation(assumeyes bool, host string, port string, connections int) bool {
+	var executeBool = false
+	if assumeyes == false {
+		fmt.Println("*** WARNING ***")
+		fmt.Println("You are going to run a TCP stress check with the following arguments:")
+		fmt.Println("  - Target: " + host)
+		fmt.Println("  - TCP Port: " + port)
+		fmt.Println("  - # of concurrent connections: " + strconv.Itoa(connections))
+		reader := bufio.NewReader(os.Stdin)
+		for {
+			fmt.Println("Please, would you confirm this execution? (y/N)")
+			response, err := reader.ReadString('\n')
+			if err != nil {
+				fmt.Println("Response not processed")
+				os.Exit(1)
+			}
+			response = strings.TrimSuffix(response, "\n")
+			if stringInSlice(response, []string{"yes", "y", "YES", "Y"}) {
+				executeBool = true
+				break
+			}
+			if stringInSlice(response, []string{"no", "n", "NO", "N", ""}) {
+				executeBool = false
+				break
+			}
+		}
+	} else {
+		executeBool = true
+	}
+	return executeBool
+}
+
 func main() {
-	hostPtr := flag.String("host", "localhost", "Host you want to open tcp connections against")
+	hostPtr := flag.String("host", "", "Host you want to open tcp connections against")
 	// according to https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers, you are probably not using this
-	portPtr := flag.Int("port", 9998, "Port you want to open tcp connections against")
+	portPtr := flag.String("port", "", "Port you want to open tcp connections against")
 	numberConnectionsPtr := flag.Int("connections", 100, "Number of connections you want to open")
 	delayPtr := flag.Int("delay", 10, "Number of ms you want to sleep between each connection creation")
-
+	assumeyesPtr := flag.Bool("assume-yes", false, "Force execution without asking for confirmation")
 	flag.Parse()
 
-	runTcpConnectionsInParallel(*numberConnectionsPtr, *delayPtr, *hostPtr, *portPtr)
+	// Target host and port are mandatory to run the TCP check
+	if *hostPtr == "" || *portPtr == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	executeBool := askForUserConfirmation(*assumeyesPtr, *hostPtr, *portPtr, *numberConnectionsPtr)
+
+	port, _ := strconv.Atoi(*portPtr)
+	if executeBool == true {
```

Comment:
following the previous recommendation: if *assumeyesPtr || askForUserConfirmation